File: implementations/go/matrix_multiplication.go
package main

import (
	"fmt"
	"os"
)

const (
	SMALL  = 50
	MEDIUM = 200
	LARGE  = 500
)

func main() {
	sizes := map[string]int{"small": SMALL, "medium": MEDIUM, "large": LARGE}
	n := sizes[os.Args[1]]

	a := make([][]int64, n)
	b := make([][]int64, n)
	c := make([][]int64, n)
	for i := 0; i < n; i++ {
		a[i] = make([]int64, n)
		b[i] = make([]int64, n)
		c[i] = make([]int64, n)
		for j := 0; j < n; j++ {
			a[i][j] = int64(i + j)
			b[i][j] = int64(i - j)
		}
	}
	for i := 0; i < n; i++ {
		for j := 0; j < n; j++ {
			var acc int64
			for k := 0; k < n; k++ {
				acc += a[i][k] * b[k][j]
			}
			c[i][j] = acc
		}
	}
	fmt.Println(c[n/2][n/2])
}

File: implementations/rust/matrix_multiplication.rs
use std::env;

const SMALL: usize  = 50;
const MEDIUM: usize = 200;
const LARGE: usize  = 500;

fn main() {
    let args: Vec<String> = env::args().collect();
    let n = match args[1].as_str() {
        "small"  => SMALL,
        "medium" => MEDIUM,
        "large"  => LARGE,
        _ => { eprintln!("Unknown size: {}", args[1]); return; }
    };
    let mut a = vec![vec![0i64; n]; n];
    let mut b = vec![vec![0i64; n]; n];
    let mut c = vec![vec![0i64; n]; n];
    for i in 0..n {
        for j in 0..n {
            a[i][j] = (i + j) as i64;
            b[i][j] = i as i64 - j as i64;
        }
    }
    for i in 0..n {
        for j in 0..n {
            let mut acc: i64 = 0;
            for k in 0..n { acc += a[i][k] * b[k][j]; }
            c[i][j] = acc;
        }
    }
    println!("{}", c[n / 2][n / 2]);
}
